experiments/state/src/main/scala/org/apache/gearpump/streaming/state/system/impl/KafkaCheckpointStore.scala

```diff
@@ -34,10 +38,13 @@ import scala.util.Try
 object KafkaCheckpointStore {
   private val LOG: Logger = LogUtil.getLogger(classOf[KafkaCheckpointStore])
 
-  def apply(config: KafkaConfig, topic: String, topicExists: Boolean) = {
-    val getConsumer = () => KafkaConsumer(topic, 0, config)
-    val producer = new KafkaProducer[Array[Byte], Array[Byte]](
-      KafkaUtil.buildProducerConfig(config), new ByteArraySerializer, new ByteArraySerializer)
+  val CONSUMER_CONFIG = "consumer_config"
+  val PRODUCER_CONFIG = "producer_config"
+
+  def apply(topic: String, topicExists: Boolean, consumerConfig: ConsumerConfig, producerConfig: Properties) = {
```

Comment:
why for consumer config, using ```kafka.consumer.ConsumerConfig``` while for producer config, using ```Properties```? This is inconsistent.